cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,268 @@
+        response = self.client.get(self.view_url)
+        self.assertContains(response, 'Maintenance', status_code=200)
+
+        # Check that all the expected links appear on the index page.
+        for url in MAINTENANCE_URLS:
+            self.assertContains(response, url, status_code=200)
+
+
+@ddt.ddt
+class MaintenanceViewTestCase(ModuleStoreTestCase):
+    """
+    Base class for maintenance view tests.
+    """
+    view_url = ''
+
+    def setUp(self):
+        super(MaintenanceViewTestCase, self).setUp()
+        self.user = AdminFactory()
+        login_success = self.client.login(username=self.user.username, password='test')
+        self.assertTrue(login_success)
+
+    def verify_error_message(self, data, error_message):
+        """
+        Verify the response contains error message.
+        """
+        response = self.client.post(self.view_url, data=data)
+        self.assertContains(response, error_message, status_code=200)
+
+    def validate_success_from_response(self, response, success_message):
+        """
+        Validate response contains success.
+        """
+        self.assertNotContains(response, '<div class="error">', status_code=200)
+        self.assertContains(response, success_message, status_code=200)
+
+    def tearDown(self):
+        """
+        Reverse the setup.
+        """
+        self.client.logout()
+        super(MaintenanceViewTestCase, self).tearDown()
+
+
+@ddt.ddt
+class MaintenanceViewAccessTests(MaintenanceViewTestCase):
+    """
+    Tests for access control of maintenance views.
+    """
+    @ddt.data(MAINTENANCE_URLS)
+    @ddt.unpack
+    def test_require_login(self, url):
+        """
+        Test that maintenance app requires user login.
+        """
+        # Log out then try to retrieve the page
+        self.client.logout()
+        response = self.client.get(url)
+
+        # Expect a redirect to the login page
+        redirect_url = '{login_url}?next={original_url}'.format(
+            login_url=reverse('login'),
+            original_url=url,
+        )
+
+        self.assertRedirects(response, redirect_url)
+
+    @ddt.data(MAINTENANCE_URLS)
+    @ddt.unpack
+    def test_global_staff_access(self, url):
+        """
+        Test that all maintenance app views are accessible to global staff user.
+        """
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    @ddt.data(MAINTENANCE_URLS)
+    @ddt.unpack
+    def test_non_global_staff_access(self, url):
+        """
+        Test that all maintenance app views are not accessible to non-global-staff user.
+        """
+        user = UserFactory(username='test', email='test@example.com', password='test')
+        login_success = self.client.login(username=user.username, password='test')
+        self.assertTrue(login_success)
+
+        response = self.client.get(url)
+        self.assertContains(response, 'Must be edX staff to perform this action.', status_code=403)
```

Comment:
this test will fail if `settings.PLATFORM_NAME` isn't `edX`. I don't know if that's a problem, tho